ADR/propagate-pull-secrets.md

```diff
@@ -0,0 +1,54 @@
+# 0. Support deploying images from Private Quay.io repositories
+
+Date: 2023-03-16
+
+## Status
+
+Provisional
+
+## Context
+
+* As a user, I would expect StoneSoup to generate a private Quay.io image repository if my source code repository is  private.
+* As a user, I would expect StoneSoup to gracefully deploy images that were pushed to private Quay.io repositories in every relevant environment.
+
+
+## Decision
+
+### Overview
+
+1. This proposal builds on the capabilties of the [image-controller](https://redhat-appstudio.github.io/book/book/image-controller.html) which sets up a Quay.io image repository per `Component` and 'downloads' a robot account token into the user's namespace for builds to be able to push to it.
+
+2. This proposal builds on the capabilities of SPI's `RemoteSecret` API. See https://github.com/redhat-appstudio/book/pull/70/files
+
+
+#### When a new `Component` is created
+
+1. image-controller creates a new Quay.io image repository
+2. image-controller provisions a new robot account token that would only be able to "pull" the `Component`'s image.
+3. image-controller creates a `Secret` in the user's workspace namespace containing the robot account token
+2. The image-controller labels the `Secret` with the name `Application`/`Component` the `Secret` is associated with. 
+
+
+#### When an expression of intent to deploy an `Application` to an `Environment` is made.
+
+
+1. Watch for creation of new `SnapshotEnvironmentBindings`.
+2. Query the 'pull' `Secret` generated by the image-controller. 
+3. Label the `Secret` with the name of the `Environment` ( ex, `spi.appstudio.redhat.com/target-environment: prod`) the `Secret` needs to be copied to, based on the `SnapshotEnvironmentBinding` CR.
```

Comment:
My proposal had started with looking out for `SnapshotEnvironmentBinding` because that shows the intent to deploy - having a Component/Application doesn't quite show that intent :-)  Thoughts?